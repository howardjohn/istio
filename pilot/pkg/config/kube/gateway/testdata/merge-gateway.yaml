apiVersion: gateway.networking.k8s.io/v1beta1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: parent
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  infrastructure:
    allowedChildren: {}
  listeners:
  - name: default
    hostname: "parent.example.com"
    port: 80
    protocol: HTTP
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: child1
  namespace: istio-system
spec:
  gatewayClassName: istio
  infrastructure:
    attachTo:
      name: parent
  listeners:
  - name: child1
    hostname: "child.example.com"
    port: 80
    protocol: HTTP
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: http
  namespace: istio-system
spec:
  parentRefs:
  - name: child1
    namespace: istio-system
  hostnames: ["child.example.com", "parent.example.com"]
  rules:
  - backendRefs:
    - name: httpbin
      port: 80
