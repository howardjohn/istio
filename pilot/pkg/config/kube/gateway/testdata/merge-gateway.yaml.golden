apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/parent/default.istio-system
  creationTimestamp: null
  name: parent-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/parent.example.com
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/parent/child1.istio-system
  creationTimestamp: null
  name: parent-istio-autogenerated-k8s-gateway-child1
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/child.example.com
    port:
      name: default
      number: 80
      protocol: HTTP
---
